packages/@glimmer/syntax/lib/generation/print.ts

```diff
@@ -57,7 +57,7 @@ export default function build(ast) {
     }
     break;
     case 'MustacheCommentStatement': {
-      output.push(compactJoin(['{{!', ast.value, '}}']));
+      output.push(compactJoin(['{{!--', ast.value, '--}}']));
```

Comment:
if that necessary? seems like it works fine without them unless you use curlys inside the comment. maybe we should adjust the parser to keep that information in the AST and only add them when they had been used before too.